### DISCUSSION TITLE
Clarify project status

### DISCUSSION THREAD

**Discussion by prymitive** [klauspost/compress#617] [Q&A] [answered]
Are you still trying to get this library changes merged upstream? I found an old ticket - #3  - but that's a few years now.
I found this library to offer a decent CPU savings (31 -> 30 cpu cores on  a busy server) when using it in Prometheus - https://github.com/prometheus/prometheus/pull/10782, so I wanted to propose using it there by default.
But this opens a questions:
- how stable is this code considered to be?
- are there any trade-offs to consider?
- is there an effort to merge those changes upstream?

Would you mind adding a note to the README with some information for everyone's benefit? Thanks!

**Comment by klauspost** [comment, marked as answer]
@prymitive 

> Are you still trying to get this library changes merged upstream? 

Mostly no. I sent in I sent in a [PR](https://github.com/golang/go/pull/38324) that gives big speedups in inflate decompression speed 2 years ago. Getting what is basically a rewrite of the compression algos in, I am not going to spend the time cleaning up the code for that.

Snappy has no interest in anything that modifies the output. So I will not pursue that either.

> how stable is this code considered to be?

It is used by hundreds of projects. Fuzz tested. I am fairly confident in the code. You should always have tests for your own integration, so you can double check for regressions.

> are there any trade-offs to consider?

Too wide of a question. Assuming you refer to deflate, it is balanced much better than stdlib. Default is about 50% the speed of the fastest mode, so it typically compresses a bit less than stdlib. Also default is "level 5" instead of level 6.

You can compare speed and compression vs stdlib [in this spreadsheet](https://docs.google.com/spreadsheets/d/1nuNE2nPfuINCZJRMt6wFWhKpToF95I47XjSsc-1rbPQ/edit?usp=sharing)

> Would you mind adding a note to the README with some information for everyone's benefit? Thanks!

This information is AFAICT already there.
